dlib/cuda/cublas_dlibapi.cpp

```diff
@@ -101,55 +101,118 @@ namespace dlib
             const tensor& lhs,
             bool trans_lhs,
             const tensor& rhs,
-            bool trans_rhs
+            bool trans_rhs,
+            size_t g_mode
```

Comment:
```suggestion
            gemm_mode g_mode
```
Why not use the enum here?

Reply:
Yeah definitely use enums instead of ints :)

Reply:
I'm proposing to add a global enum in tensor_tools.h that will serve the various functions needing to differentiate their internal calculations based on tensor organization:
`enum class operation_mode { CHANNEL_WISE = 0, PLANE_WISE = 1 };`
I'll update the code accordingly to ensure clarity and consistency.